### PR TITLE
Undo redo manager batch 1

### PR DESCRIPTION
**Pull request by scribblemaniac**
These are the suggested changes I have so far for the undo-redo-manager branch. This should just about cover all the issues I've found outside of backupmanager.h/.cpp and backupelement.h/.cpp. Review of those files is ongoing and will require some back-and-forth.

### CODE REVIEW

File: core_lib/src/interface/editor.cpp [hunk 1]
```diff
@@ -1077,86 +1058,60 @@ KeyFrame* Editor::addNewKey()
...
+
     if (!layer->keyExistsWhichCovers(frameIndex))
     {
         return;
     }
 
+    if (shouldBackup)
+    {
+        backup(tr("Remove frame"));
+    }
```
Comment (MrStevns) [inline core_lib/src/interface/editor.cpp:1091]: Why would you want to remove a key and not do a state change? we won't be able to go back otherwise.

**Review by MrStevns — APPROVED** [review]
I have one comment but otherwise the changes looks good.
👍 

I will merge.

File: core_lib/src/interface/editor.cpp (hunk 1 above)
Comment (scribblemaniac) [reply core_lib/src/interface/editor.cpp:1091]: removeKeyAtLayerId wasn't backing it up before these changes. You want to remove a key and not do a state change when you're calling that from backupelements so you don't get into some infinite backup loop. Key removals triggered by an action use removeKey() which still does a back.

Comment (MrStevns) [reply core_lib/src/interface/editor.cpp:1091]: Agh.. you're right, I forgot about that one. I couldn't remember it well enough and the diff didn't make it obvious.

**Pull request merged**